database/postgres/build_list.go

```diff
@@ -159,6 +159,8 @@ func (c *client) GetRepoBuildList(r *library.Repo, filters map[string]interface{
 	err = c.Postgres.
 		Table(constants.TableBuild).
 		Where("repo_id = ?", r.GetID()).
+		Where("started < ?", before).
```

Comment:
Something to check: Can you look at this query out in production to see how it behaves from a performance and cost perspective? It's probably ok assuming there's an index on `repo_id` which would narrow the rows down a lot, but since there's not an index on `started`, I'm curious how fast those queries and what their cost is.

I'm assuming you know how to grab the generated query.?  And then you could throw that in a query tool and run `explain analyze $your_query`.  That'll show how long it takes and its cost.  It's probably fine, especially if people don't use this often, but as the database grows over time, this kind of stuff can add up and need to be considered for database tuning and maintenance.

Reply:
I agree that performance should be explored and validated. A LIMIT doesn't do much if it's paired up with an ORDER statement since the DB will still have to do a full sort on that ORDER column before it can even consider LIMITing the return rows. If you didn't have the order, then a limit on its own would probably be fine, but we have them both together.

https://dba.stackexchange.com/questions/110636/poor-performance-on-query-with-limit-when-i-add-an-order-by

Reply:
We indeed have `repo_id` as an index in the DB. I ran a few test queries as well as the `explain analyse` tool. It seems like the query is fairly performant. I got results that varied from a couple tenths of a millisecond to two milliseconds.

Reply:
It's probably worth calling out the [indexes](https://github.com/go-vela/server/blob/master/database/postgres/ddl/repo.go#L41-L47) are visible from the application code.